<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>SPI</class>
 <widget class="QWidget" name="SPI">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>735</width>
    <height>300</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Form</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <property name="spacing">
    <number>5</number>
   </property>
   <property name="leftMargin">
    <number>1</number>
   </property>
   <property name="topMargin">
    <number>1</number>
   </property>
   <property name="rightMargin">
    <number>1</number>
   </property>
   <property name="bottomMargin">
    <number>1</number>
   </property>
   <item>
    <widget class="QScrollArea" name="scrollArea">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="minimumSize">
      <size>
       <width>0</width>
       <height>50</height>
      </size>
     </property>
     <property name="maximumSize">
      <size>
       <width>16777215</width>
       <height>16777215</height>
      </size>
     </property>
     <property name="widgetResizable">
      <bool>true</bool>
     </property>
     <widget class="QWidget" name="scrollAreaWidgetContents">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>0</y>
        <width>896</width>
        <height>230</height>
       </rect>
      </property>
      <layout class="QHBoxLayout" name="horizontalLayout_9">
       <item>
        <widget class="QLabel" name="label_6">
         <property name="font">
          <font>
           <family>Monospace</family>
           <pointsize>9</pointsize>
          </font>
         </property>
         <property name="text">
          <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-size:9pt; font-weight:600;&quot;&gt;spi [device (1/2)] [pull (up/down/floating)] [mode (master/slave)] &lt;/span&gt;&lt;/p&gt;&lt;p&gt;&lt;span style=&quot; font-size:9pt; font-weight:600;&quot;&gt;[frequency (value hz/khz/mhz)] [polarity 0/1] [phase 0/1] [msb-first/lsb-first]. &lt;/span&gt;&lt;/p&gt;&lt;table border=&quot;0&quot; style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px;&quot; cellspacing=&quot;2&quot; cellpadding=&quot;0&quot;&gt;&lt;tr&gt;&lt;td&gt;&lt;p&gt;&lt;span style=&quot; font-family:'Hack'; font-size:9pt; font-weight:600;&quot;&gt;[&lt;/span&gt;&lt;/p&gt;&lt;/td&gt;&lt;td&gt;&lt;p&gt;&lt;span style=&quot; font-size:9pt;&quot;&gt;Chip select (CS) active (low).&lt;/span&gt;&lt;/p&gt;&lt;/td&gt;&lt;/tr&gt;&lt;tr&gt;&lt;td&gt;&lt;p&gt;&lt;span style=&quot; font-family:'Hack'; font-size:9pt; font-weight:600;&quot;&gt;]&lt;/span&gt;&lt;/p&gt;&lt;/td&gt;&lt;td&gt;&lt;p&gt;&lt;span style=&quot; font-size:9pt;&quot;&gt;CS disable (high).&lt;/span&gt;&lt;/p&gt;&lt;/td&gt;&lt;/tr&gt;&lt;tr&gt;&lt;td&gt;&lt;p&gt;&lt;span style=&quot; font-family:'Hack'; font-size:9pt; font-weight:600;&quot;&gt;r&lt;/span&gt;&lt;/p&gt;&lt;/td&gt;&lt;td&gt;&lt;p&gt;&lt;span style=&quot; font-size:9pt;&quot;&gt;Read one byte by sending dummy byte (&lt;/span&gt;&lt;span style=&quot; font-family:'Hack'; font-size:9pt;&quot;&gt;0xff&lt;/span&gt;&lt;span style=&quot; font-size:9pt;&quot;&gt;). &lt;/span&gt;&lt;span style=&quot; font-family:'Hack'; font-size:9pt;&quot;&gt;r:1…255&lt;/span&gt;&lt;span style=&quot; font-size:9pt;&quot;&gt; for bulk reads.&lt;/span&gt;&lt;/p&gt;&lt;/td&gt;&lt;/tr&gt;&lt;tr&gt;&lt;td&gt;&lt;p&gt;&lt;span style=&quot; font-family:'Hack'; font-size:9pt; font-weight:600;&quot;&gt;hd&lt;/span&gt;&lt;/p&gt;&lt;/td&gt;&lt;td&gt;&lt;p&gt;&lt;span style=&quot; font-size:9pt;&quot;&gt;Read one byte by sending dummy byte (&lt;/span&gt;&lt;span style=&quot; font-family:'Hack'; font-size:9pt;&quot;&gt;0xff&lt;/span&gt;&lt;span style=&quot; font-size:9pt;&quot;&gt;). &lt;/span&gt;&lt;span style=&quot; font-family:'Hack'; font-size:9pt;&quot;&gt;hd:1…4294967295&lt;/span&gt;&lt;span style=&quot; font-size:9pt;&quot;&gt; for bulk reads. Displays a hexdump of the result.&lt;/span&gt;&lt;/p&gt;&lt;/td&gt;&lt;/tr&gt;&lt;tr&gt;&lt;td&gt;&lt;p&gt;&lt;span style=&quot; font-family:'Hack'; font-size:9pt; font-weight:600;&quot;&gt;w&lt;/span&gt;&lt;/p&gt;&lt;/td&gt;&lt;td&gt;&lt;p&gt;&lt;span style=&quot; font-size:9pt;&quot;&gt;Followed by values to write byte(s). &lt;/span&gt;&lt;span style=&quot; font-family:'Hack'; font-size:9pt;&quot;&gt;w:1…255&lt;/span&gt;&lt;span style=&quot; font-size:9pt;&quot;&gt; for bulk writes.&lt;/span&gt;&lt;/p&gt;&lt;/td&gt;&lt;/tr&gt;&lt;tr&gt;&lt;td&gt;&lt;p&gt;&lt;span style=&quot; font-family:'Hack'; font-size:9pt; font-weight:600;&quot;&gt;0b&lt;/span&gt;&lt;/p&gt;&lt;/td&gt;&lt;td&gt;&lt;p&gt;&lt;span style=&quot; font-size:9pt;&quot;&gt;Write this binary value. Format is &lt;/span&gt;&lt;span style=&quot; font-family:'Hack'; font-size:9pt;&quot;&gt;0b00000000&lt;/span&gt;&lt;span style=&quot; font-size:9pt;&quot;&gt; for a byte, but partial bytes are also fine: &lt;/span&gt;&lt;span style=&quot; font-family:'Hack'; font-size:9pt;&quot;&gt;0b1001&lt;/span&gt;&lt;span style=&quot; font-size:9pt;&quot;&gt;.&lt;/span&gt;&lt;/p&gt;&lt;/td&gt;&lt;/tr&gt;&lt;tr&gt;&lt;td&gt;&lt;p&gt;&lt;span style=&quot; font-family:'Hack'; font-size:9pt; font-weight:600;&quot;&gt;0&lt;/span&gt;&lt;/p&gt;&lt;/td&gt;&lt;td&gt;&lt;p&gt;&lt;span style=&quot; font-size:9pt;&quot;&gt;Write this Octal value. Format is prefixed by a &lt;/span&gt;&lt;span style=&quot; font-family:'Hack'; font-size:9pt;&quot;&gt;0&lt;/span&gt;&lt;span style=&quot; font-size:9pt;&quot;&gt; (values from &lt;/span&gt;&lt;span style=&quot; font-family:'Hack'; font-size:9pt;&quot;&gt;000&lt;/span&gt;&lt;span style=&quot; font-size:9pt;&quot;&gt; to &lt;/span&gt;&lt;span style=&quot; font-family:'Hack'; font-size:9pt;&quot;&gt;077&lt;/span&gt;&lt;span style=&quot; font-size:9pt;&quot;&gt;)&lt;/span&gt;&lt;/p&gt;&lt;/td&gt;&lt;/tr&gt;&lt;tr&gt;&lt;td&gt;&lt;p&gt;&lt;span style=&quot; font-family:'Hack'; font-size:9pt; font-weight:600;&quot;&gt;0h/0x&lt;/span&gt;&lt;/p&gt;&lt;/td&gt;&lt;td&gt;&lt;p&gt;&lt;span style=&quot; font-size:9pt;&quot;&gt;Write this HEX value. Format is &lt;/span&gt;&lt;span style=&quot; font-family:'Hack'; font-size:9pt;&quot;&gt;0h01&lt;/span&gt;&lt;span style=&quot; font-size:9pt;&quot;&gt; or &lt;/span&gt;&lt;span style=&quot; font-family:'Hack'; font-size:9pt;&quot;&gt;0x01&lt;/span&gt;&lt;span style=&quot; font-size:9pt;&quot;&gt;. Partial bytes are fine: &lt;/span&gt;&lt;span style=&quot; font-family:'Hack'; font-size:9pt;&quot;&gt;0xA&lt;/span&gt;&lt;span style=&quot; font-size:9pt;&quot;&gt;. &lt;/span&gt;&lt;span style=&quot; font-family:'Hack'; font-size:9pt;&quot;&gt;A-F&lt;/span&gt;&lt;span style=&quot; font-size:9pt;&quot;&gt; can be lower-case or capital letters.&lt;/span&gt;&lt;/p&gt;&lt;/td&gt;&lt;/tr&gt;&lt;tr&gt;&lt;td&gt;&lt;p&gt;&lt;span style=&quot; font-family:'Hack'; font-size:9pt; font-weight:600;&quot;&gt;0-255&lt;/span&gt;&lt;/p&gt;&lt;/td&gt;&lt;td&gt;&lt;p&gt;&lt;span style=&quot; font-size:9pt;&quot;&gt;Write this decimal value. Any number not preceded by &lt;/span&gt;&lt;span style=&quot; font-family:'Hack'; font-size:9pt;&quot;&gt;0x&lt;/span&gt;&lt;span style=&quot; font-size:9pt;&quot;&gt;, &lt;/span&gt;&lt;span style=&quot; font-family:'Hack'; font-size:9pt;&quot;&gt;0h&lt;/span&gt;&lt;span style=&quot; font-size:9pt;&quot;&gt;, or &lt;/span&gt;&lt;span style=&quot; font-family:'Hack'; font-size:9pt;&quot;&gt;0b&lt;/span&gt;&lt;span style=&quot; font-size:9pt;&quot;&gt; is interpreted as a decimal value.&lt;/span&gt;&lt;/p&gt;&lt;/td&gt;&lt;/tr&gt;&lt;tr&gt;&lt;td&gt;&lt;p&gt;&lt;span style=&quot; font-family:'Hack'; font-size:9pt; font-weight:600;&quot;&gt;&amp;quot;&lt;/span&gt;&lt;/p&gt;&lt;/td&gt;&lt;td&gt;&lt;p&gt;&lt;span style=&quot; font-size:9pt;&quot;&gt;Write an ASCII-encoded string&lt;/span&gt;&lt;/p&gt;&lt;/td&gt;&lt;/tr&gt;&lt;/table&gt;&lt;/body&gt;&lt;/html&gt;</string>
         </property>
         <property name="textFormat">
          <enum>Qt::RichText</enum>
         </property>
         <property name="scaledContents">
          <bool>false</bool>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
   <item>
    <layout class="QHBoxLayout" name="SPI_mode_lay">
     <property name="leftMargin">
      <number>1</number>
     </property>
     <property name="topMargin">
      <number>1</number>
     </property>
     <property name="rightMargin">
      <number>1</number>
     </property>
     <property name="bottomMargin">
      <number>1</number>
     </property>
     <item>
      <widget class="QComboBox" name="cSPI_dev">
       <item>
        <property name="text">
         <string>device 1</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>device 2</string>
        </property>
       </item>
      </widget>
     </item>
     <item>
      <widget class="QComboBox" name="cSPI_pull">
       <item>
        <property name="text">
         <string>up</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>down</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>floating</string>
        </property>
       </item>
      </widget>
     </item>
     <item>
      <widget class="QComboBox" name="cSPI_role">
       <item>
        <property name="text">
         <string>master</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>slave</string>
        </property>
       </item>
      </widget>
     </item>
     <item>
      <widget class="QComboBox" name="cSPI_freq">
       <item>
        <property name="text">
         <string>320000</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>650000</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>1310000</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>2620000</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>5250000</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>10500000</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>21000000</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>42000000</string>
        </property>
       </item>
      </widget>
     </item>
     <item>
      <widget class="QComboBox" name="cSPI_polph">
       <item>
        <property name="text">
         <string>polarity 0 phase 0</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>polarity 0 phase 1</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>polarity 1 phase 0</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>polarity 1 phase 1</string>
        </property>
       </item>
      </widget>
     </item>
     <item>
      <widget class="QComboBox" name="cSPI_msb">
       <item>
        <property name="text">
         <string>msb-first</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>lsb-first</string>
        </property>
       </item>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="bSPI_Connect">
       <property name="text">
        <string>RUN SPI</string>
       </property>
       <property name="checkable">
        <bool>true</bool>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <widget class="QFrame" name="SPI_frame">
     <property name="minimumSize">
      <size>
       <width>0</width>
       <height>100</height>
      </size>
     </property>
     <property name="frameShape">
      <enum>QFrame::StyledPanel</enum>
     </property>
     <property name="frameShadow">
      <enum>QFrame::Raised</enum>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_2">
      <property name="spacing">
       <number>5</number>
      </property>
      <property name="leftMargin">
       <number>0</number>
      </property>
      <property name="topMargin">
       <number>0</number>
      </property>
      <property name="rightMargin">
       <number>0</number>
      </property>
      <property name="bottomMargin">
       <number>0</number>
      </property>
      <item>
       <layout class="QHBoxLayout" name="SPI_CMD_lay">
        <property name="leftMargin">
         <number>1</number>
        </property>
        <property name="topMargin">
         <number>1</number>
        </property>
        <property name="rightMargin">
         <number>1</number>
        </property>
        <property name="bottomMargin">
         <number>1</number>
        </property>
        <item>
         <widget class="QPushButton" name="bSPI_pinOut">
          <property name="text">
           <string>Pinout</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="bSPI_NSS">
          <property name="minimumSize">
           <size>
            <width>50</width>
            <height>0</height>
           </size>
          </property>
          <property name="text">
           <string>NSS</string>
          </property>
          <property name="checkable">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="bSPI_Write">
          <property name="text">
           <string>WRITE</string>
          </property>
          <property name="autoRepeat">
           <bool>true</bool>
          </property>
          <property name="autoRepeatDelay">
           <number>1000</number>
          </property>
          <property name="autoRepeatInterval">
           <number>500</number>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLineEdit" name="eSPI_singleWrite">
          <property name="minimumSize">
           <size>
            <width>50</width>
            <height>0</height>
           </size>
          </property>
          <property name="inputMask">
           <string/>
          </property>
          <property name="placeholderText">
           <string>[ 0x80:4 ]</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="bSPI_Read">
          <property name="minimumSize">
           <size>
            <width>50</width>
            <height>0</height>
           </size>
          </property>
          <property name="text">
           <string>READ</string>
          </property>
          <property name="autoRepeat">
           <bool>true</bool>
          </property>
          <property name="autoRepeatDelay">
           <number>1000</number>
          </property>
          <property name="autoRepeatInterval">
           <number>500</number>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLineEdit" name="eSPI_singleRead">
          <property name="minimumSize">
           <size>
            <width>50</width>
            <height>0</height>
           </size>
          </property>
          <property name="inputMask">
           <string/>
          </property>
          <property name="text">
           <string/>
          </property>
          <property name="readOnly">
           <bool>false</bool>
          </property>
          <property name="placeholderText">
           <string>[ r:64 ]</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="SPI_Dump_lay">
        <item>
         <widget class="QLabel" name="label_10">
          <property name="text">
           <string>Addres, hex</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLineEdit" name="eSPI_DumpAddr">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="placeholderText">
           <string>0x3F 0xA0</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="label_11">
          <property name="text">
           <string>Data len:</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QSpinBox" name="eSPI_DumpLen">
          <property name="maximum">
           <number>16777216</number>
          </property>
          <property name="value">
           <number>128</number>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="label_12">
          <property name="text">
           <string>NSS mode</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QComboBox" name="eSPI_NSSmode">
          <property name="currentIndex">
           <number>1</number>
          </property>
          <item>
           <property name="text">
            <string>no</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>full pack</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>rx</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>tx</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>rx and tx with gap</string>
           </property>
          </item>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="label_13">
          <property name="text">
           <string>Command Line: </string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLineEdit" name="eSPI_DumpCmdOut">
          <property name="readOnly">
           <bool>true</bool>
          </property>
          <property name="placeholderText">
           <string>[ 0x3F 0xA0 %:10 r:128 ]</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="bSPI_Dump">
          <property name="text">
           <string>DUMP</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>bSPI_Dump</sender>
   <signal>clicked()</signal>
   <receiver>SPI</receiver>
   <slot>cmdDump()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>691</x>
     <y>274</y>
    </hint>
    <hint type="destinationlabel">
     <x>367</x>
     <y>149</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>bSPI_Connect</sender>
   <signal>clicked(bool)</signal>
   <receiver>SPI</receiver>
   <slot>setSPI(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>685</x>
     <y>179</y>
    </hint>
    <hint type="destinationlabel">
     <x>367</x>
     <y>149</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>bSPI_Read</sender>
   <signal>clicked()</signal>
   <receiver>SPI</receiver>
   <slot>cmdRead()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>494</x>
     <y>223</y>
    </hint>
    <hint type="destinationlabel">
     <x>367</x>
     <y>149</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>bSPI_Write</sender>
   <signal>clicked()</signal>
   <receiver>SPI</receiver>
   <slot>cmdWrite()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>212</x>
     <y>223</y>
    </hint>
    <hint type="destinationlabel">
     <x>367</x>
     <y>149</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>bSPI_NSS</sender>
   <signal>clicked(bool)</signal>
   <receiver>SPI</receiver>
   <slot>cmdNSS(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>127</x>
     <y>223</y>
    </hint>
    <hint type="destinationlabel">
     <x>367</x>
     <y>149</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>bSPI_pinOut</sender>
   <signal>clicked()</signal>
   <receiver>SPI</receiver>
   <slot>cmdPinout()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>42</x>
     <y>223</y>
    </hint>
    <hint type="destinationlabel">
     <x>367</x>
     <y>149</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>cSPI_dev</sender>
   <signal>currentIndexChanged(int)</signal>
   <receiver>SPI</receiver>
   <slot>setDevice(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>49</x>
     <y>179</y>
    </hint>
    <hint type="destinationlabel">
     <x>367</x>
     <y>149</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>setSPI(bool)</slot>
  <slot>setDevice(int)</slot>
  <slot>cmdRead()</slot>
  <slot>cmdWrite()</slot>
  <slot>cmdPinout()</slot>
  <slot>cmdNSS(bool)</slot>
  <slot>cmdDump()</slot>
 </slots>
</ui>
